### ISSUE TITLE
Last.fm / scrobbling support

### ISSUE THREAD

**Issue by iparr** [iina/iina#2172]
I'd like to make a feature request:
- Support for scrobbling to [last.fm](https://www.last.fm/api) and…
- Libre.fm (I'm presuming that work towards one would benefit the other)

I've just trialed this application and - aside from this feature - it seems like a very good replacement for my current audio-only player.

Best of luck with the project and keep up the good work.

**Comment by inflation** [comment]
You can do it by writing a plug-in. We want to keep the player as universal as possible. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by patrickclery** [comment]
> @inflation Could you provide any documentation on how a plug-in is implemented for Mac OS.
> 
> I'm an iOS developer so would love to learn new stuff about Mac OS development.

Here's a list of last.fm API clients in different languages: https://www.last.fm/api/downloads

**Comment by skywinder** [comment]
@inflation 1) is there any list of plugins available? 2) is there docs, how to vuild plugin?

**Comment by gingerbeardman** [comment]
I was just about to file this issue:




**What you want IINA to do:**
I'd like IINA to support audio scrobbling (sending currently playing track info to last.fm)
 

**What IINA does currently:**
n/a


**Why you think this should be added:**
Many users like to send what they listen to last.fm

This is possible with many other players, see below.

I open folders of audio files in IINA as a lightweight music player.


**Examples of other projects that have something similar:**
- VLC

- Clementine
- Deezer
- HypeMachine
- MusicBee
- Sonos
- Spotify
- Swinsian
- Tidal
- 8tracks

- Details https://www.last.fm/about/trackmymusic
- Other services by using third party apps https://getsatisfaction.com/lastfm/topics/more-ways-to-scrobble

API: https://www.last.fm/api/scrobbling

**Comment by gingerbeardman** [comment]
> We want to keep the player as universal as possible.

Can you expand on this? i don't understand, especially given that cross-platform player VLC supports scrobbling

**Comment by ethanndickson** [comment]
> You can do it by writing a plug-in. We want to keep the player as universal as possible.
> 
> Sent with [GitHawk](http://githawk.com)

I couldn't find any scrobbling plug-ins, so I just made one that runs off an mpv event.

 https://github.com/ethanndickson/iina-lastfm.

Let me know if you find any issues.

**Comment by dertuxmalwieder** [comment]
Ah, Right. IINA uses mpv. Good approach! I hope I’ll be able to test it by the weekend.

**Comment by iparr** [comment]
I'm using this now and it seems to work really well.

Great work, @ethanndickson

**Comment by amanessinger** [comment]
Yeah, works fine. A link in the README should be added, and then this issue can be closed for good.
